sklearn/decomposition/_nmf.py

```diff
@@ -903,6 +903,11 @@ def non_negative_factorization(X, W=None, H=None, n_components=None, *,
 
         - None: 'nndsvd' if n_components < n_features, otherwise 'random'.
 
+        .. deprecated:: 0.24
+           the `init` value, when 'init=None' and if
+           n_components <= min(n_samples, n_features) will be changed from
+           `'nndsvd'` to `'nndsvda'` in 0.26.
```

Comment:
As Tom said, it's not a deprecation. So I think the proper way to do that is to add nothing here for now and then add a `.. versionchanged` in 2 versions.